gui/model_download_manager.py

```
@@ -520,6 +521,28 @@ def run(self):
+
+        # Pre-flight disk-space check so a multi-GB download fails fast with a
+        # clear message instead of dying mid-write — the parallel path even
+        # pre-allocates the full size, which can sparse-succeed and only fail
+        # later as opaque write errors.
+        if total:
+            probe_dir = self.target_dir
+            while not probe_dir.exists() and probe_dir != probe_dir.parent:
+                probe_dir = probe_dir.parent
+            try:
+                free = shutil.disk_usage(probe_dir).free
+            except Exception:
```

Ruff (0.15.18)

[warning] Do not catch blind exception: `Exception`

(BLE001)